fhttp/http_test.go

```diff
@@ -283,6 +286,38 @@ func TestParseStatus(t *testing.T) {
 	}
 }
 
+func TestParseDelay(t *testing.T) {
+	var tests = []struct {
+		input    string
+		expected time.Duration
+	}{
+		// Error cases
+		{"", -1},
+		{"x", -1},
+		{"1::", -1},
+		{"x:10", -1},
+		{"10ms:-1", -1},
+		{"20ms:101", -1},
+		{"20ms:101%", -1},
+		{"10ms:45,100ms:56", -1},
+		// Max delay case:
+		{"10s:45,10s:55", 1 * time.Second},
+		// Good cases
+		{"100ms", 100 * time.Millisecond},
+		{"100ms:100", 100 * time.Millisecond},
+		{"100ms:100%", 100 * time.Millisecond},
```

Comment:
curious: why allow '%' in the delay string to be valid?

Reply:
it's the probability percentage - adding the optional % makes it a bit more self documented (I thought, but clearly not ;-) )

Reply:
i understood the purpose. i was just wondering if it is worth the hassle.